dbos/workflow.go

```diff
@@ -436,219 +537,238 @@ func runAsWorkflow[P any, R any](ctx context.Context, fn WorkflowFunc[P, R], inp
 	// The buffer size of 1 allows the goroutine to send the outcome without blocking
 	// In addition it allows the channel to be garbage collected
-	outcomeChan := make(chan workflowOutcome[R], 1)
-
-	// Create the handle
-	handle := &workflowHandle[R]{
-		workflowID:  workflowStatus.ID,
-		outcomeChan: outcomeChan,
-	}
+	outcomeChan := make(chan workflowOutcome[any], 1)
 
 	// Create workflow state to track step execution
 	wfState := &workflowState{
-		workflowID:  workflowStatus.ID,
-		stepCounter: -1,
+		workflowID: workflowID,
+		stepID:     -1,
 	}
 
 	// Run the function in a goroutine
-	augmentUserContext := context.WithValue(ctx, workflowStateKey, wfState)
-	dbos.workflowsWg.Add(1)
+	workflowCtx := WithValue(c, workflowStateKey, wfState)
+	c.workflowsWg.Add(1)
 	go func() {
-		defer dbos.workflowsWg.Done()
-		result, err := fn(augmentUserContext, input)
+		defer c.workflowsWg.Done()
+		result, err := fn(workflowCtx, input)
 		status := WorkflowStatusSuccess
 		if err != nil {
 			status = WorkflowStatusError
 		}
-		recordErr := dbos.systemDB.UpdateWorkflowOutcome(dbosWorkflowContext, updateWorkflowOutcomeDBInput{workflowID: workflowStatus.ID, status: status, err: err, output: result})
+		recordErr := c.systemDB.UpdateWorkflowOutcome(c.ctx, updateWorkflowOutcomeDBInput{
+			workflowID: workflowID,
+			status:     status,
+			err:        err,
+			output:     result,
+		})
 		if recordErr != nil {
-			outcomeChan <- workflowOutcome[R]{result: *new(R), err: recordErr}
+			outcomeChan <- workflowOutcome[any]{result: nil, err: recordErr}
 			close(outcomeChan) // Close the channel to signal completion
 			return
 		}
-		outcomeChan <- workflowOutcome[R]{result: result, err: err}
+		outcomeChan <- workflowOutcome[any]{result: result, err: err}
 		close(outcomeChan) // Close the channel to signal completion
 	}()
 
-	// Run the peer goroutine to handle cancellation and timeout
-	/*
-		if dbosWorkflowContext.Done() != nil {
-			getLogger().Debug("starting goroutine to handle workflow context cancellation or timeout")
-			go func() {
-				select {
-				case <-dbosWorkflowContext.Done():
-					// The context was cancelled or timed out: record timeout or cancellation
-					// CANCEL WORKFLOW HERE
-					return
-				}
-			}()
-		}
-	*/
+	// Commit the transaction
+	if err := tx.Commit(c.ctx); err != nil {
+		return nil, newWorkflowExecutionError(workflowID, fmt.Sprintf("failed to commit transaction: %v", err))
+	}
 
-	return handle, nil
+	return &workflowHandle[any]{workflowID: workflowID, outcomeChan: outcomeChan, dbosContext: c}, nil
 }
 
 /******************************/
 /******* STEP FUNCTIONS *******/
 /******************************/
 
-type StepFunc[P any, R any] func(ctx context.Context, input P) (R, error)
+type StepFunc func(ctx context.Context, input any) (any, error)
+type GenericStepFunc[P any, R any] func(ctx context.Context, input P) (R, error)
+
+const StepParamsKey DBOSContextKey = "stepParams"
```

Comment:
This is how users can set step parameters, without passing functional options nor a new input parameter.